packages/react-inbox/src/components/Messages/index.tsx

```diff
@@ -18,39 +20,47 @@ const Messages: React.FunctionComponent<InboxProps> = ({
   renderMessage,
   unreadCount,
 }) => {
-  useMessages();
-  const { messages, isLoading } = useInbox();
+  const containerRef = useRef<HTMLDivElement>();
+  const { atBottom, reset } = useAtBottom(containerRef.current);
+  const {
+    messages, isLoading, fetchMore,
+  } = useMessages();
+
+  useEffect(() => {
+    if (atBottom && !isLoading) {
+      fetchMore();
+      reset();
+    }
+  }, [atBottom, fetchMore, isLoading, reset]);
   return (
     <>
       {renderHeader ? (
         renderHeader({})
       ) : (
         <Header data-testid="header">
-          <HeaderText>{title}{unreadCount ? ` (${unreadCount})` : ''}</HeaderText>
-          <BodyText style={{cursor: 'pointer'}}>Mark all as read</BodyText>
+          <HeaderText>{title}{unreadCount ? ` (${unreadCount})` : ""}</HeaderText>
+          <BodyText style={{ cursor: "pointer" }}>Mark all as read</BodyText>
         </Header>
       )}
       <TabBar />
-      <Body data-testid="messages">
-        {isLoading ? (
-          <Loading />
-        ) : (
-          messages?.map((message) =>
-            renderMessage ? (
-              renderMessage(message)
-            ) : (
-              <Message key={message.messageId} {...message} />
-            )
-          )
-        )}
+      <Body ref={containerRef as React.RefObject<HTMLDivElement>} data-testid="messages">
+        {messages?.map((message) =>
+          renderMessage ? (
+            renderMessage(message)
+          ) : (
+            <Message key={message.messageId} {...message} />
+          ),
+        )
+        }
+        {isLoading && <Loading />}
         {!isLoading && messages?.length === 0 && (
           <Empty>You have no notifications at this time</Empty>
         )}
       </Body>
       {renderFooter ? renderFooter({}) : (
-      <Footer>
-        <div><span style={{marginTop: 2}}>Powered by&nbsp;&nbsp;</span><CourierLogo /></div>
-      </Footer>)}
+        <Footer>
```

Comment:
this should prob be the default "renderFooter" function so people can override it

Reply:
we can do this later